cwms/timeseries/timeseries.py

```diff
@@ -7,6 +7,90 @@
 from cwms.types import JSON, Data
 
 
+def update_timeseries_groups(
+    group_id: str, office_id: str, replace_assigned_ts: Optional[bool] = False
+) -> None:
+    """
+    Updates the timeseries groups with the provided group ID and office ID.
+
+    Parameters
+    ----------
+    group_id : str
+        The new specified timeseries ID that will replace the old ID.
+    office_id : str
+        The ID of the office associated with the specified level.
+    replace_assigned_ts : bool, optional
+        Specifies whether to unassign all existing time series before assigning new time series specified in the content body. Default is False.
+
+    Returns
+    -------
+    None
+    """
+    if not group_id:
+        raise ValueError("Cannot update a specified level without an id")
+    if not office_id:
+        raise ValueError("Cannot update a specified level without an office id")
+
+    endpoint = f"timeseries/group/{group_id}"
+    params = {
+        "replace-assigned-ts": replace_assigned_ts,
+        "office": office_id,
+    }
+
+    # Assuming api.patch is a valid function available in your environment
+    api.patch(endpoint=endpoint, params=params)
+
+
+def timeseries_group_df_to_json(
+    data: pd.DataFrame,
+    group_id: str,
+    office_id: str,
+) -> JSON:
+    """
+    Converts a dataframe to a json dictionary in the correct format.
+
+    Parameters
+    ----------
+    data: pd.DataFrame
+        Dataframe containing timeseries information.
+    group_id: str
+        The group ID for the timeseries.
+
+    Returns
+    -------
+    json
+        JSON dictionary of the timeseries data.
+    """
+    required_columns = ["office-id", "ts-id", "alias", "ts-code", "attribute"]
```

Comment:
ts-id and office-id should be the only required columns i the dataframe.  for your purpose you will also be passing in alias, but that isn't required in this function.  you should check if it is present in the dataframe and if not create a column and set it to None.  if attribute is not present set it to 0 and is ts-code is not present just don't include it in the json output.  it isn't needed.